### ISSUE TITLE
Expand `log` to `console.log()`

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/feedback#750]
### Check for existing issues

- [X] Completed

### Is your feature request related to a problem?

In JetBrains’ tooling, in JS and TS, you can type `log` then hit `tab` and it will expand to `console.log()`.  It’s a small thing, but it’s so nice.  We should see if we can add this feature.  Maybe this calls for some generic system that allows you to define, per language, maps that link some input text into an expanded form.

### Describe the solution you'd like

-

**Comment by Pringlers** [comment]
We need generalized version of this. So each user can add their own shortcut to boost productivity 🚀

**Comment by aexvir** [comment]
yep, live templates and postfixes are amazing features from the intellij platform, and if zed would implement something similar that would be an amazing productivity boost

I'm attaching a couple screenshots from both live templates, which can be invoked directly or if there is a `$SELECTION$` block they can also be used to wrap code in refactors; and postfix expressions, which allows configuring custom templates to replace the code when invoking `.expression`

@ushankax @Pringlers it would be nice to 👍🏻 this issue if you think this is a nice feature 🙂 

![capture-20221214093358](https://user-images.githubusercontent.com/8055505/207547894-38f04ae5-010d-4677-9371-e2ae364a9a79.png)
![capture-20221214093304](https://user-images.githubusercontent.com/8055505/207547897-6d784477-6195-49c9-9e70-7f3f4a4059e4.png)

**Comment by rkusa** [comment]
It might be possible to achieve this already with an LSP snippet, similar to https://github.com/zed-industries/feedback/issues/566#issuecomment-1273791513
